src/context.cpp

```diff
@@ -415,6 +418,47 @@ namespace librealsense
     }
 
 #ifdef BUILD_WITH_DDS
+    class dds_sensor_proxy : public software_sensor
+    {
+    public:
+        dds_sensor_proxy( std::string name, software_device* owner,
+                          std::shared_ptr< dds::dds_device > const & dev, size_t index) :
+            software_sensor(name, owner),
+            _dev(dev),
+            _index(index)
+        {
+        }
+
+        void open( const stream_profiles & profiles ) override
+        {
+            std::vector< rs2_video_stream > rs2_profs;
+            for (size_t i = 0; i < profiles.size(); ++i)
+            {
+                rs2_profs[i].fps = profiles[i]->get_framerate();
+                rs2_profs[i].fmt = profiles[i]->get_format();
+                rs2_profs[i].type = profiles[i]->get_stream_type();
+                const auto&& vsp = As<video_stream_profile, stream_profile_interface>( profiles[i] );
```

Comment:
Interesting, what does `As` do?

Reply:
This doesn't work?
```
auto vsp = profiles[i].as< rs2::video_stream_profile >();
```

Reply:
stream_profile_interface doesn't have a `as` method. `As` is a general function in `extension.h` that uses `dynamic_cast`